src/modules/sor/order.ts

```diff
@@ -0,0 +1,53 @@
+import { BatchSwap, SwapInfo, Swaps, SwapType } from '@balancer-labs/sdk';
+import { ADDRESSES } from '@/constants/addresses';
+import { convertSwapInfoToBatchSwap } from './batch-swap';
+import { getSorSwap, orderKindToSwapType } from './sor';
+import { SorRequest, SorOrderResponse, PriceResponse } from './types';
+
+/**
+ * Takes a SOR request and returns a SorOrder which contains a transaction that
+ * the user can send to the chain.
+ */
+export async function createSorOrder(
+  networkId: number,
+  request: SorRequest
+): Promise<SorOrderResponse> {
+  validateSorRequest(request);
+
+  const swapInfo: SwapInfo = await getSorSwap(networkId, request);
+  const swapType: SwapType = orderKindToSwapType(request.orderKind);
+  const batchSwap: BatchSwap = convertSwapInfoToBatchSwap(
+    swapType,
+    swapInfo,
+    request.sender,
+    request.recipient || request.sender,
+    request.slippagePercentage
+  );
+  const encodedBatchSwapData = Swaps.encodeBatchSwap(batchSwap);
+
+  const priceResponse: PriceResponse = {
+    sellAmount: swapInfo.swapAmount,
+    buyAmount: swapInfo.returnAmount,
+    allowanceTarget: ADDRESSES[networkId].contracts.vault,
+    price: swapInfo.marketSp,
+  };
+
+  return {
+    price: priceResponse,
+    to: ADDRESSES[networkId].contracts.vault,
+    data: encodedBatchSwapData,
+    value: '0',
```

Comment:
What if the user is trading ETH for DAI? Would the value not have to be specified here?

Reply:
True, this doesn't support ETH swaps yet. Will fix in a future PR.